apps/token/src/components/nav/nav-dropdown.tsx

```diff
@@ -0,0 +1,206 @@
+  <DropdownMenuPrimitive.Content
+    {...contentProps}
+    ref={forwardedRef}
+    className="min-w-[290px] bg-neutral-200 dark:bg-neutral-900 mt-4 p-2 rounded-xl border border-neutral-700"
+    align="start"
+    sideOffset={10}
+  />
+));
+
+/**
+ * The component that contains the dropdown menu items.
+ */
+export const DropdownMenuItem = forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.Item>,
+  React.ComponentProps<typeof DropdownMenuPrimitive.Item>
+>(({ className, ...itemProps }, forwardedRef) => (
+  <DropdownMenuPrimitive.Item
+    {...itemProps}
+    ref={forwardedRef}
+    className={classNames(itemClass, className)}
+  />
+));
+
+/**
+ * An item that can be controlled and rendered like a checkbox.
+ */
+export const DropdownMenuCheckboxItem = forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.CheckboxItem>,
+  React.ComponentProps<typeof DropdownMenuPrimitive.CheckboxItem>
+>(({ className, ...checkboxItemProps }, forwardedRef) => (
+  <DropdownMenuPrimitive.CheckboxItem
+    {...checkboxItemProps}
+    ref={forwardedRef}
+    className={classNames(itemClass, className)}
+  />
+));
+
+/**
+ * An item that can be controlled and rendered like a radio.
+ */
+export const DropdownMenuRadioItem = forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.RadioItem>,
+  React.ComponentProps<typeof DropdownMenuPrimitive.RadioItem> & {
+    inset?: boolean;
+  }
+>(({ className, inset = false, ...radioItemprops }, forwardedRef) => (
+  <DropdownMenuPrimitive.RadioItem
+    {...radioItemprops}
+    ref={forwardedRef}
+    className={classNames(itemClass, className)}
+  />
+));
+
+/**
+ * Renders when the parent {@link DropdownMenuCheckboxItem} or {@link DropdownMenuRadioItem} is checked.
+ * You can style this element directly, or you can use it as a wrapper to put an icon into, or both.
+ */
+export const DropdownMenuItemIndicator = forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.ItemIndicator>,
+  React.ComponentProps<typeof DropdownMenuPrimitive.ItemIndicator>
+>(({ ...itemIndicatorProps }, forwardedRef) => (
+  <DropdownMenuPrimitive.ItemIndicator
+    {...itemIndicatorProps}
+    ref={forwardedRef}
+    className="flex-end"
+  >
+    <Icon name="tick" />
+  </DropdownMenuPrimitive.ItemIndicator>
+));
+
+/**
+ * Used to visually separate items in the dropdown menu.
+ */
+export const DropdownMenuSeparator = forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.Separator>,
+  React.ComponentProps<typeof DropdownMenuPrimitive.Separator>
+>(({ className, ...separatorProps }, forwardedRef) => (
+  <DropdownMenuPrimitive.Separator
+    {...separatorProps}
+    ref={forwardedRef}
+    className={classNames(
+      'h-px my-1 mx-2 bg-neutral-700 dark:bg-black',
+      className
+    )}
+  />
+));
+
+export const NavDropDown = ({ navbarTheme }: { navbarTheme: NavbarTheme }) => {
+  const { t } = useTranslation();
+  const [isOpen, setOpen] = useState(false);
+  const subRoutes = [
```

Comment:
Suggestion: I think we'd get some value from this config being separated out. We could then use it to not only build this menu, but also improve the cypress common function for page navigation